### PR TITLE
Changes to DriverInterface

### PR DESCRIPTION
**Pull request by nealsid**
#446 Fix some type inconsistencies and replace calls to IOKitCallStructMethod with IOKitCallScalarMethod.

### CODE REVIEW

File: src/topologyentry.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022-, Intel Corporation
+
+#ifndef TOPOLOGY_ENTRY_HEADER
```
Comment (rdementi) [inline src/topologyentry.h:4]: we should use "#pragma once" going forward instead of #ifndef include guards

**Review by rdementi — DISMISSED** [review]
sorry for delay. I have just one minor suggestion

File: src/topologyentry.h (hunk 1 above)
Comment (nealsid) [reply src/topologyentry.h]: Done,

**Comment by nealsid** [comment]
Thanks!

**Review by opcm — APPROVED** [review]
thanks!

**Pull request merged**